Make HTTP progress visible at the information level

joelverhagen opened the pull request:
Per @davidfowl's request, show the HTTP operations during the restore at the information (default) logging level.

@emgarten @anurse

joelverhagen committed:
Make HTTP progress visible by default

dnfclas commented:
Hi __@joelverhagen__, I'm your friendly neighborhood .NET Foundation Pull Request Bot (You can call me DNFBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by .NET Foundation and real humans are currently evaluating your PR.</p>

TTYL, DNFBOT;

analogrelay commented:
We might actually want to keep these messages as `Verbose` and just make verbose the default and have a `--quiet` setting to set the default back to `Info`. We do still want a way to suppress the HTTP progress but keep the other status messages.

joelverhagen commented:
@anurse, sounds good. I'll talked to @emgarten about this.

analogrelay commented:
FYI: The specific switch we use to enable a quiet-mode doesn't really worry me. Keeping `--verbosity` is fine, as long as there is a way to make it quieter :)

joelverhagen commented:
Spoke to @emgarten offline about this. Added an additional log level called "minimal", which is between information and warning. The default log is information level. I move all currently information logs to minimal and moved the HTTP status logs to information.

The equivalent to DNU `--quiet` is `--verbosity minimal`, which no longer shows the HTTP progress.

There is currently a hack in the Visual Studio `OnBuildPackageRestorer` progress popup code that ignores information (and now minimal) level logs if they start with a space. This should be cleaned up so that logs starting with a space are logged as information and only minimal is observed by the VS code. This will eliminate the need for this magic.

davidfowl commented:
Screenshot?

emgarten commented on src/NuGet.Core/NuGet.Protocol.Core.v3/RemoteRepositories/RemoteV2FindPackageByIdResource.cs:
```diff
@@ -179,7 +179,7 @@ into nextLink
                 catch (Exception ex) when (retry < 2)
                 {
-                    Logger.LogInformation(string.Format("Warning: FindPackagesById: {1}\r\n  {0}", ex.Message, id));
+                    Logger.LogMinimal(string.Format("Warning: FindPackagesById: {1}\r\n  {0}", ex.Message, id));
```
It seems like these should be at the warning level

emgarten commented on src/NuGet.Core/NuGet.Commands/RestoreCommand.cs:
```diff
@@ -80,7 +80,7 @@ public Task<RestoreResult> ExecuteAsync()
                 relockFile = true;
                 _request.ExistingLockFile.IsLocked = false;
-                _logger.LogInformation(Strings.Log_LockFileOutOfDate);
+                _logger.LogMinimal(Strings.Log_LockFileOutOfDate);
```
This should be a warning

emgarten commented on src/NuGet.Core/NuGet.CommandLine.XPlat/CommandOutputLogger.cs:
```diff
@@ -79,18 +85,21 @@ private void LogInternal(LogLevel logLevel, string message)
                         caption = "info ";
                         break;
+                    case LogLevel.Minimal:
+                        caption = "mimim";
```
what if we just called this `"log  "` or `"output"` ?

joelverhagen replied:
Fixed in latest. Rebased off of latest `dev`.

joelverhagen commented:
Examples:
- [Default logging when restoring `dotnet new`](https://gist.github.com/joelverhagen/c3c1dc4d0479040faea7/d8de3c32ff20a7a8e5aafb44fa7b11f8e5d32a69)
- [`--verbosity minimal`](https://gist.github.com/joelverhagen/c3c1dc4d0479040faea7/8bb270ad98f3557611a5aa5893db3367fc2f3bd6)

emgarten commented:
:shipit:
